server/src/internalClusterTest/java/org/opensearch/snapshots/CloneSnapshotIT.java

```diff
@@ -153,6 +161,86 @@ public void testCloneSnapshotIndex() throws Exception {
         assertEquals(status1.getStats().getTotalSize(), status2.getStats().getTotalSize());
     }
 
+    public void testCloneShallowSnapshotIndex() throws Exception {
```

Comment:
Can you also add test for below:
Scenario 1:
1. Create a full copy snapshot with `remoteStoreIndexShallowCopy` flag as FALSE in the repository
2. Update the repo setting to set flag `remoteStoreIndexShallowCopy` as TRUE.
3. Clone the snapshot created in step 1.
4. Verify that cloned snapshot is also created with full copy and should not have any lock files

Scenario 2:
1. Create a shallow copy snapshot with `remoteStoreIndexShallowCopy` flag as TRUE in the repository
2. Update the repo setting to set flag `remoteStoreIndexShallowCopy` as false
3. Clone the snapshot created in step 1.
4. Verify that cloned snapshot is also created as shallow copy snapshot and should have 2 lock files

Reply:
Added these tests in the new revision